packages/langchain/src/orchestration/client.ts

```diff
@@ -212,8 +232,18 @@ export class OrchestrationClient extends BaseChatModel<
       }
     };
     config.templating = this.orchestrationConfig.templating;
-    if (config.templating && isTemplate(config.templating) && tools.length) {
-      config.templating.tools = [...(config.templating.tools || []), ...tools];
+    if (tools.length) {
+      if (!config.templating) {
+        config.templating = {};
+      }
+      if (config.templating && !isTemplateRef(config.templating)) {
```

Comment:
Since you set the config to an empty object if it is not present, doesn't the first part of this check become redundant?

Reply:
I assume it's because typescript has issues understanding that it's never undefined, but in that case here I would use `!`